homeassistant/components/edl21/config_flow.py

```diff
@@ -0,0 +1,62 @@
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import voluptuous as vol
+from .sensor import DOMAIN, CONF_SERIAL_PORT
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+from typing import Any
+
+import serial.tools.list_ports
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+
+class EDL21ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for EDL21."""
+
+    VERSION = 1
+
+    async def async_step_pick_manual(self, user_input: dict[str, Any]) -> FlowResult:
+        return self.async_create_entry(title="EDL21", data=user_input)
+
+    async def async_step_user(self, user_input: dict[str, Any]) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is not None:
+            if user_input[CONF_SERIAL_PORT] == CONF_MANUAL_PATH:
+                return self.async_show_form(
```

Comment:
move this into the manual step and call that one instead (separation of concerns).